### ISSUE TITLE
Coverage of bash script fails on cat<<EOF || foo statement

### ISSUE THREAD

**Issue by neg** [SimonKagstrom/kcov#50]
I agree that the construct is bad and should never be used, but I'm working on a project where it's heavily used...

```bash
#!/bin/bash

# This works
cat<<EOF
foo
bar
EOF

echo "This also works"

# This breaks
cat<<EOF || echo "cat failed"
foo
bar
EOF

echo "From here on it do not work!"
```

**Comment by SimonKagstrom** [comment]
Should be pretty simple to fix. The bug is here:

https://github.com/SimonKagstrom/kcov/blob/master/src/engines/bash-engine.cc#L356

so kcov will try to find {{{EOF || echo "cat failed"}}} as the end-marker.


I guess there should be a real parser for shell scripts to know what is actual code, but I was hoping to get by without that :-)

**Comment by SimonKagstrom** [comment]
Fixed with 21ba53f, thanks for the report!
